src/Data/YAML/Event/Writer.hs

```diff
@@ -192,8 +251,9 @@ putNode = \docMarker -> go (-1 :: Int) (not docMarker) BlockIn
 
       Plain -- empty scalars
         | t == "", Nothing <- anc, Tag Nothing <- tag -> contEol -- not even node properties
-        | sol, t == "" ->             anchorTag0 anc tag (if c == BlockKey then ws <> cont else contEol)
-        | t == "", BlockKey <- c   -> anchorTag0 anc tag (ws <> cont)
+        | sol, t == "" ->             anchorTag0 anc tag (if  c == BlockKey || c == FlowKey then ws <> cont else contEol)
```

Comment:
we should reorder this to have `t == ""` first to be consistent with all the other guards; and we might consider factoring out the common `t == ""` condition altogether to make it more obvious what is going on here; as in

```hs
      Plain -- empty scalars
        | t == "" -> case () of
                            _ | Nothing <- anc, Tag Nothing <- tag -> contEol
                              | sol -> ...
                              | BlockKey <- c  -> ...
                              | FlowKey <- c -> ...
                              | otherwise -> anchorTag'' (Left ws) anc tag contEol
```